### PR TITLE
Add support for optics

### PR DESCRIPTION
**Pull request by arybczak**

### CODE REVIEW

File: insert-ordered-containers.cabal [hunk 1]
```diff
@@ -46,6 +46,7 @@ library
     , base-compat           >=0.10.5  && <0.12
     , hashable              >=1.2.6.1 && <1.4
     , lens                  >=4.17    && <4.19
+    , optics                >=0.2     && <0.3
```
Comment (phadej) [inline insert-ordered-containers.cabal]: `optics-extra` would be enough, wouldn't it?

Comment (arybczak) [reply insert-ordered-containers.cabal]: optics-core + optics-extra. Should I change it?

Comment (phadej) [reply insert-ordered-containers.cabal]: Yes please. Slightly less dependencies: good.

(In fact I thought of splitting out the `lens` (and `optics`) stuff, but that could wait).

File: insert-ordered-containers.cabal [hunk 2]
```diff
@@ -46,6 +43,8 @@ library
     , base-compat           >=0.10.5  && <0.12
```
Comment (phadej) [inline insert-ordered-containers.cabal]: - `base` should be bumped to `>=4.9`
- also `transformers` lower bump can be bumped and CPP cleaned up. Dropping old GHC support should be fine by now.

`swagger2` would be affected, but it should be fine https://github.com/GetShopTV/swagger2/issues/199


I'll do that after merge & before release

Comment (arybczak) [reply insert-ordered-containers.cabal]: Alright. I also have a corresponding swagger2 PR that adds support for optics btw, that's what prompted this PR ;)

Comment (phadej) [reply insert-ordered-containers.cabal]: I'm curious to see swagger2 PR.

Comment (phadej) [reply insert-ordered-containers.cabal]: Too fast enter.

I'll make a release today.

Does `swagger2` change introduce `LabelOptic` instances for everything, so you don't have problems with name clashes? I.e. `optics` usage is 

```haskell
mempty
   & #type .~ SwaggerString
```

?

If it's so, than I'll support the change. It's quite neat that way.

Comment (arybczak) [reply insert-ordered-containers.cabal]: Yes, that's exactly how it looks.

**Comment by phadej** [comment]
released https://hackage.haskell.org/package/insert-ordered-containers-0.2.3

**Pull request merged**